### ISSUE TITLE
Error / Exception Question

### ISSUE THREAD

**Issue by markparr** [christiandelbianco/monitor-table-change-with-sqltabledependency#177]
I opened issue #176 recently in which I was seeing StatusChanges events firing but was not seeing Change events.  Finally found my problem today in that one of my DB fields is defined as an nchar(16) and was being populated with a "numeric text" value generated by the stored procedure that adds records to the table.  In my application, I had the item defined as an INT in my model mapping table.  Whenever I had values for the table column within the valid range for INT, the OnChange event would fire.

I assume some kind of exception / error was getting thrown and caught at some level but it was never passed to my application either as an exception or within my OnError event handler.  Should this data mismatch exception have been logged someplace or thrown as an exception?

Thanks
Mark

**Comment by christiandelbianco** [comment]
Hi. I committed a test case where i reproduce your case. My table is:

```
CREATE TABLE [dbo].[Issue117](
	[Id] [nchar](16) NULL,
	[Message] [nvarchar](50) NULL
) 
```
and C# model is:

```
        private class Issue177Model
        {
            public int Id { get; set; }
            public string Message { get; set; }
        }

```
If you run the **Issue177Test** you will see that everything goes fine.

**Comment by markparr** [comment]
Is there supposed to be a link to Issue177Test?  If so it is not highlighting as such,

**Comment by christiandelbianco** [comment]
i push this test case in the repository. you can find there.

**Comment by markparr** [comment]
Christian, 

I found and your test example and if I am reading it correctly, you use values of 1 and 1234 for ID. 
 Manually populate your DB w/ an ID string value of '1234567890123456' and see if you get notified in your test application.  

My program was getting notified w/ "small" values being added to my table but it was when I got up in "large values" that would not convert to an INT that I did not get notified or any error notification.

Mark

**Comment by christiandelbianco** [comment]
thank you for the clarification. I'll see what I can do. 

However, keep in mind that there is the possibility that in case of _impossible cast,_ i will throws an exception when SqlTableDependency will starts, as the C# model type and the Table column type do not match. I think that this is the most correct way of handling this type of mismatch.

**Comment by markparr** [comment]
Please let us know.  I agree an exception or OnError event that can be identified and handled programmatically in an application is the best option over logging.  I would have identified my typo sooner with it.  

Thanks
Mark

**Comment by christiandelbianco** [comment]
**OnError** event is raise and SqlTableDependency stop working.
Fix released with version  8.5.7
Regards
